AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLDocument/GraphQLDocument+Subscription.swift

```diff
@@ -16,11 +16,14 @@ public struct GraphQLSubscription: GraphQLDocument {
     public let documentType = GraphQLDocumentType.subscription
     public let modelType: Model.Type
     public let subscriptionType: GraphQLSubscriptionType
+    public let syncEnabled: Bool
 
     public init(of modelType: Model.Type,
-                type subscriptionType: GraphQLSubscriptionType) {
+                type subscriptionType: GraphQLSubscriptionType,
+                syncEnabled: Bool = false) {
```

Comment:
I'm not sure if it makes sense to expose a syncEnabled flag for this class, right?  Seems like if we have a subscription with a model, syncEnabled has to be true, right?

Reply:
a backend can be provisioned with/without sync so the graphql document can be constructed with sync related fields or without by using this `syncEnabled` parameter